sweepbatcher/sweep_batch.go

```diff
@@ -566,6 +580,54 @@ func (b *batch) addSweep(ctx context.Context, sweep *sweep) (bool, error) {
 		}
 	}
 
+	// If presigned mode is enabled, we should first presign the new version
+	// of batch transaction. Also ensure that all the sweeps in the batch
+	// use the same mode (presigned or regular).
+	if sweep.presigned {
+		// Ensure that all the sweeps in the batch use presigned mode.
+		for _, s := range b.sweeps {
+			if !s.presigned {
+				b.log().Infof("failed to add sweep %x to the "+
+					"batch, because the batch has "+
+					"non-presigned sweep %x",
+					sweep.swapHash[:6], s.swapHash[:6])
+
+				return false, nil
+			}
+		}
+
+		if len(b.sweeps) != 0 {
+			if err := b.presign(ctx, sweep); err != nil {
+				b.log().Infof("failed to add sweep %x to the "+
+					"batch, because failed to presign new "+
+					"version of batch tx: %v",
+					sweep.swapHash[:6], err)
+
+				return false, nil
+			}
+		} else {
+			if err := b.ensurePresigned(ctx, sweep); err != nil {
```

Comment:
if `err == nil` here, don't we then have to call `b.presign` for the sweep here?

Reply:
If `ensurePresigned` returned no error, this means, it has been presigned already. No need to presign again. If not presigned, then crash the batcher (this is only possible because of a bug).